python_modules/dagster/dagster/_core/instance/runs/run_domain.py

```diff
@@ -720,11 +728,29 @@ def get_keys_to_reexecute(
             {key for key in to_reexecute if isinstance(key, AssetCheckKey)},
         )
 
+    def _get_repo_scoped_asset_node(
```

Comment:
we should leave some explainer maybe by this function "we get repo scoped asset nodes via the context instead of the repo directly to leverage request contexts potential for caching" or smth